pilot/cmd/pilot-agent/options/options.go

```diff
@@ -102,6 +102,9 @@ var (
 	// DNSCaptureByAgent is a copy of the env var in the init code.
 	DNSCaptureByAgent = env.Register("ISTIO_META_DNS_CAPTURE", false,
 		"If set to true, enable the capture of outgoing DNS packets on port 53, redirecting to istio-agent on :15053")
+	// EnableDNSAtGateway enables DNS server at Gateways.
+	EnableDNSAtGateway = env.Register("ISTIO_META_ENABLE_DNS_SERVER", false,
+		"If set to true, starts the DNS server on :15053. This can be used when we want Gateways to use DNS proxy")
```

Comment:
This maybe confusing, by default gateway cannot use it. We have no iptables set for it, right

Reply:
That is why I called this `ISTIO_META_ENABLE_DNS_SERVER` - so it just starts the DNS server - it does not capture the DNS traffic whoever wants (Dynamic Forward proxy in our example) can point to this resolver

Reply:
I mean you can mention in the desription

Reply:
ok. .updated. PTAL